cluster/gce/windows/configure.ps1

```diff
@@ -159,6 +158,8 @@ try {
 
   $config = New-FileRotationConfig
   Schedule-LogRotation -Pattern '.*\.log$' -Path ${env:LOGS_DIR} -RepetitionInterval $(New-Timespan -Hour 1) -Config $config
+
+  Pull-InfraContainer
```

Comment:
What's the reason for this change?

On one hand I suppose this should let the node join the cluster more quickly. On the other hand there could be some delay if a user tries to run a Windows pod immediately after the node is Ready but the pause container hasn't finished pulling yet.

Reply:
it is not needed for kubelet/node to register with the master/control plane. Pull-InfraContainer is a best-effort optimization, so that when a user start a pod, the 'pause' image is likely already present on the node. Moving it to the last would 1) shorten the delay for the node to join the cluster, [so user can submit deployment/pod earlier, user does not need to wait for the node to be ready to submit deployment/pod in theory, but in practice, user may wait/check for node to be ready before deploying anything] 2) no worse than current situation (should be better) in terms of user-perceived node startup time. [we may explore moving it to the first step when making it async/pulling in the background]

Reply:
Makes sense, thanks for clarifying.

Reply:
It doesn't seem safe to move this.

Windows image pull is slow, for docker pulling the pause image takes ~4min, for containerd it is even longer...

If the pause image is not prepulled, and the node becomes ready, it is possible that a user will never be able to create a pod.

The reason is that in kubelet we set a 4min timeout for pod sandbox start (including pulling the pause image) https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/remote/remote_runtime.go#L97.

If the pause image is not prepulled, `RunPodSandbox` may timeout every time, abort the ongoing pulling, and try pulling again. If for any reason the pause image pull is longer than 4min, the image pull may be always aborted, thus the pod can't start any more.

For Docker, I believe this may happen, and for containerd, because of the slow image pull this breaks the test.

Reply:
> If the pause image is not prepulled, RunPodSandbox may timeout every time, abort the ongoing pulling, and try pulling again.

This doesn't sound like the right thing to do on Windows. Can/should we change that behavior?

Reply:
chatted offline about it.